fix python-apollo tests in travis build

https://travis-ci.org/github/GMOD/Apollo/jobs/694742617

2 fails:

```
======================================================================
FAIL: test_get_group_creator (test.group_test.GroupTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/GMOD/Apollo/python-apollo/test/group_test.py", line 57, in test_get_group_creator
    assert user_info['username'] == "admin@local.host"
AssertionError: 
>>  assert {'firstName': 'Junior', 'lastName': 'Galaxy', 'inactive': False, 'role': 'USER', 'availableGroups': [{'name': 'one_group'}, {'name': 'another_group'}, {'name': 'temp_group'}], 'userCount': 3, 'searchName': None, 'groups': [], 'userId': 2, 'organismPermissions': [{'organism': 'test_organism', 'permissions': ['WRITE', 'EXPORT', 'READ'], 'permissionArray': ['WRITE', 'EXPORT', 'READ'], 'id': 5, 'userId': 2}, {'organism': 'alt_org', 'permissions': ['WRITE', 'EXPORT', 'READ'], 'permissionArray': ['WRITE', 'EXPORT', 'READ'], 'id': 6, 'userId': 2}, {'organism': 'org3', 'permissions': ['WRITE', 'EXPORT', 'READ'], 'permissionArray': ['WRITE', 'EXPORT', 'READ'], 'id': 7, 'userId': 2}, {'organism': 'org4', 'permissions': ['WRITE', 'EXPORT', 'READ'], 'permissionArray': ['WRITE', 'EXPORT', 'READ'], 'id': 8, 'userId': 2}], 'username': 'test@bx.psu.edu'}['username'] == "admin@local.host"
```

and

```
======================================================================
FAIL: test_get_group_admin (test.group_test.GroupTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/GMOD/Apollo/python-apollo/test/group_test.py", line 65, in test_get_group_admin
    assert user_info['username'] == "admin@local.host"
AssertionError: 
>>  assert {'firstName': 'Junior', 'lastName': 'Galaxy', 'inactive': False, 'role': 'USER', 'availableGroups': [{'name': 'one_group'}, {'name': 'another_group'}, {'name': 'temp_group'}], 'userCount': 3, 'searchName': None, 'groups': [], 'userId': 2, 'organismPermissions': [{'organism': 'test_organism', 'permissions': ['WRITE', 'EXPORT', 'READ'], 'permissionArray': ['WRITE', 'EXPORT', 'READ'], 'id': 5, 'userId': 2}, {'organism': 'alt_org', 'permissions': ['WRITE', 'EXPORT', 'READ'], 'permissionArray': ['WRITE', 'EXPORT', 'READ'], 'id': 6, 'userId': 2}, {'organism': 'org3', 'permissions': ['WRITE', 'EXPORT', 'READ'], 'permissionArray': ['WRITE', 'EXPORT', 'READ'], 'id': 7, 'userId': 2}, {'organism': 'org4', 'permissions': ['WRITE', 'EXPORT', 'READ'], 'permissionArray': ['WRITE', 'EXPORT', 'READ'], 'id': 8, 'userId': 2}], 'username': 'test@bx.psu.edu'}['username'] == "admin@local.host"
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -51,33 +51,7 @@ cache:
     - $HOME/.npm/
     - $HOME/.gradle/caches/
     - $HOME/.gradle/wrapper/
-script:
-  - |
-    if [[ $TEST_SUITE == "apollo" ]]; then
-      travis_wait ./gradlew installJBrowseWebOnly gwtc installJBrowseTools && ./grailsw refresh-dependencies --stacktrace && ./grailsw test-app -coverage -xml --stacktrace && ./grailsw coveralls
-    fi
-  - |
-    if [[ $TEST_SUITE == "javascript" ]]; then
-      jshint client/apollo/js
-    fi
-  - |
-    if [[ $TEST_SUITE == "python-apollo" ]]; then
-      set -ex
-      cp test/config/python-apollo.travis apollo-config.groovy
-      ./grailsw run-app &
-      git clone --single-branch --branch fix-default-apollo-test --depth=1 https://github.com/galaxy-genome-annotation/python-apollo
-      cd python-apollo
-      sed -i -e 's|8080/|8080/apollo|' `pwd`/test-data/local-arrow.yml
-      export ARROW_GLOBAL_CONFIG_PATH=`pwd`/test-data/local-arrow.yml
-      echo `pwd`/test-data/local-arrow.yml
-      python3 -m venv .venv
-      . .venv/bin/activate
-      pip3 install .
-      pip3 install nose
-      ./bootstrap_apollo.sh --nodocker
-      python3 setup.py nosetests
-      killall java || true
-    fi
+script: "./test/config/run-travis.sh"
 
 # jobs:
 #   allow_failures:

--- a/grails-app/controllers/org/bbop/apollo/UserController.groovy
+++ b/grails-app/controllers/org/bbop/apollo/UserController.groovy
@@ -86,12 +86,12 @@ class UserController {
 
 
             def users = c.list(max: maxResults, offset: offset) {
-                if (dataObject.userId && dataObject.userId.isLong()) {
+                if (dataObject.userId && (dataObject instanceof Number || dataObject.userId.isLong())) {
                     eq('id', dataObject.userId as Long)
                 }
                 else
-                if (dataObject.userId && dataObject.userId as String) {
-                    eq('username', dataObject.userId)
+                if (dataObject.userId) {
+                    eq('username', dataObject.userId as String)
                 }
                 if (searchName) {
                     or {

--- a/grails-app/controllers/org/bbop/apollo/UserController.groovy
+++ b/grails-app/controllers/org/bbop/apollo/UserController.groovy
@@ -86,7 +86,7 @@ class UserController {
 
 
             def users = c.list(max: maxResults, offset: offset) {
-                if (dataObject.userId && (dataObject instanceof Number || dataObject.userId.isLong())) {
+                if (dataObject.userId && (dataObject.userId instanceof Number || dataObject.userId.isLong())) {
                     eq('id', dataObject.userId as Long)
                 }
                 else

--- a/grails-app/controllers/org/bbop/apollo/UserController.groovy
+++ b/grails-app/controllers/org/bbop/apollo/UserController.groovy
@@ -82,11 +82,15 @@ class UserController {
             def showInactiveUsers = dataObject.showInactiveUsers ?: false
             def omitEmptyOrganisms = dataObject.omitEmptyOrganisms != null ? dataObject.omitEmptyOrganisms : false
 
+//            log.debug "data object ${dataObject as JSON} . . . ${dataObject.userId.isInteger()} ${dataObject.userId.isLong() }"
+
+
             def users = c.list(max: maxResults, offset: offset) {
-                if (dataObject.userId && dataObject.userId in Integer) {
-                    eq('id', (Long) dataObject.userId)
+                if (dataObject.userId && dataObject.userId.isLong()) {
+                    eq('id', dataObject.userId as Long)
                 }
-                if (dataObject.userId && dataObject.userId in String) {
+                else
+                if (dataObject.userId && dataObject.userId as String) {
                     eq('username', dataObject.userId)
                 }
                 if (searchName) {

--- a/test/config/run-travis.sh
+++ b/test/config/run-travis.sh
@@ -41,7 +41,9 @@ if [[ $TEST_SUITE == "python-apollo" ]]; then
       . .venv/bin/activate
       pip3 install .
       pip3 install nose
+      echo "Running bootstrap_apollo nodocker"
       ./bootstrap_apollo.sh --nodocker
+      echo "Running nosetests"
       python3 setup.py nosetests
       killall java || true
 fi

--- a/test/config/run-travis.sh
+++ b/test/config/run-travis.sh
@@ -5,7 +5,7 @@ EXIT_STATUS=0
 echo "Running test $TEST_SUITE"
 
 if [[ $TEST_SUITE == "apollo" ]]; then
-  travis_wait ./gradlew installJBrowseWebOnly gwtc installJBrowseTools && ./grailsw refresh-dependencies --stacktrace && ./grailsw test-app -coverage -xml --stacktrace && ./grailsw coveralls
+  ./gradlew installJBrowseWebOnly gwtc installJBrowseTools && ./grailsw refresh-dependencies --stacktrace && ./grailsw test-app -coverage -xml --stacktrace && ./grailsw coveralls
 fi
 if [[ $TEST_SUITE == "javascript" ]]; then
   jshint client/apollo/js

--- a/test/config/run-travis.sh
+++ b/test/config/run-travis.sh
@@ -30,8 +30,8 @@ if [[ $TEST_SUITE == "python-apollo" ]]; then
       set -ex
       cp test/config/python-apollo.travis apollo-config.groovy
       ./grailsw run-app &
-#      git clone --single-branch --branch fix-default-apollo-test --depth=1 https://github.com/galaxy-genome-annotation/python-apollo
-      git clone --single-branch --branch 4.2.2 --depth=1 https://github.com/galaxy-genome-annotation/python-apollo
+      git clone --single-branch --branch fix-default-apollo-test --depth=1 https://github.com/galaxy-genome-annotation/python-apollo
+#      git clone --single-branch --branch 4.2.2 --depth=1 https://github.com/galaxy-genome-annotation/python-apollo
       cd python-apollo
       sed -i -e 's|8080/|8080/apollo|' "`pwd`/test-data/local-arrow.yml"
       ARROW_GLOBAL_CONFIG_PATH="`pwd`/test-data/local-arrow.yml"

--- a/test/config/run-travis.sh
+++ b/test/config/run-travis.sh
@@ -1,0 +1,49 @@
+#!/bin/bash
+
+EXIT_STATUS=0
+
+echo "Running test $TEST_SUITE"
+
+if [[ $TEST_SUITE == "apollo" ]]; then
+  travis_wait ./gradlew installJBrowseWebOnly gwtc installJBrowseTools && ./grailsw refresh-dependencies --stacktrace && ./grailsw test-app -coverage -xml --stacktrace && ./grailsw coveralls
+fi
+if [[ $TEST_SUITE == "javascript" ]]; then
+  jshint client/apollo/js
+fi
+if [[ $TEST_SUITE == "python-apollo" ]]; then
+#  set -ex
+#  cp src/integration-test/groovy/resources/travis/python-apollo.travis apollo-config.groovy
+#  ./grailsw run-app &
+#  git clone --single-branch --branch 4.2.2 --depth=1 https://github.com/galaxy-genome-annotation/python-apollo
+#  cd python-apollo
+#  sed -i 's|8888|8080/apollo|' `pwd`/test-data/local-arrow.yml
+#  export ARROW_GLOBAL_CONFIG_PATH=`pwd`/test-data/local-arrow.yml
+#  python3 --version
+#  python3 -m venv .venv
+#  . .venv/bin/activate
+#  python3 --version
+#  pip3 install .
+#  pip3 install nose
+#  ./bootstrap_apollo.sh --nodocker
+#  python3 setup.py nosetests
+#  killall java || true
+      set -ex
+      cp test/config/python-apollo.travis apollo-config.groovy
+      ./grailsw run-app &
+#      git clone --single-branch --branch fix-default-apollo-test --depth=1 https://github.com/galaxy-genome-annotation/python-apollo
+      git clone --single-branch --branch 4.2.2 --depth=1 https://github.com/galaxy-genome-annotation/python-apollo
+      cd python-apollo
+      sed -i -e 's|8080/|8080/apollo|' "`pwd`/test-data/local-arrow.yml"
+      ARROW_GLOBAL_CONFIG_PATH="`pwd`/test-data/local-arrow.yml"
+      export ARROW_GLOBAL_CONFIG_PATH
+      echo "`pwd`/test-data/local-arrow.yml"
+      python3 -m venv .venv
+      . .venv/bin/activate
+      pip3 install .
+      pip3 install nose
+      ./bootstrap_apollo.sh --nodocker
+      python3 setup.py nosetests
+      killall java || true
+fi
+
+exit $EXIT_STATUS

